src/gui/properties/speedplotview.cpp

```diff
@@ -33,15 +33,67 @@
 #include "base/global.h"
 #include "base/utils/misc.h"
 
+namespace
+{
+    enum PeriodInSeconds
+    {
+        MIN1_SEC = 60,
+        MIN5_SEC = 5 * 60,
+        MIN30_SEC = 30 * 60,
+        HOUR6_SEC = 6 * 60 * 60
+    };
+
+    const int MIN5_BUF_SIZE = 5 * 60;
+    const int MIN30_BUF_SIZE = 5 * 60;
+    const int HOUR6_BUF_SIZE = 5 * 60;
+    const int DIVIDER_30MIN = MIN30_SEC / MIN30_BUF_SIZE;
+    const int DIVIDER_6HOUR = HOUR6_SEC / HOUR6_BUF_SIZE;
+}
+
+SpeedPlotView::Averager::Averager(int divider, boost::circular_buffer<PointData> &sink)
+    : m_divider(divider)
+    , m_sink(sink)
+    , m_counter(0)
+    , m_accumulator {}
+{
+}
+
+void SpeedPlotView::Averager::push(const PointData &pointData)
+{
+    // Accumulator overflow will be hit in worst case on longest used averaging span,
+    // defined by divider value. Maximum divider is DIVIDER_6HOUR = 72
+    // Using int32 for accumulator we get overflow when transfer speed reaches 2^31/72 ~~ 28.4 MBytes/s.
+    // With quint64 this speed limit is 2^64/72 ~~ 228 PBytes/s.
+    // This speed is inaccessible to ordinary user.
```

Comment:
@dzmat:  **Grammar Alert:** 

>inaccessible to ordinary **user**. 

should be 

>inaccessible to ordinary **users**. 

or 

>inaccessible to **an** ordinary user.

Reply:
Fixed